02_C# Fundamentals/SystemWatcherApp/SystemWatcherApp/Program.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using SystemWatcherApp.Configs;
+using SystemWatcherApp.Resources;
+
+namespace SystemWatcherApp
+{
+    class Program
+    {
+        readonly static List<string> startDirectories;
+        readonly static List<FileSystemWatcher> watchers;
+        readonly static string defaultDirectory = @"D:\_02_BCL\Default\";
+        static CultureInfo currentCulture;
+        static List<RuleElement> rules;
+
+        static Program()
+        {
+            rules = new List<RuleElement>();
+            startDirectories = new List<string>();
+            watchers = new List<FileSystemWatcher>();
+        }
+
+        static void Main(string[] args)
+        {
+            Console.OutputEncoding = Encoding.UTF8;
+
+            WatcherSectionGroup watcherSectionGroup = ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None).SectionGroups["watcherSection"] as WatcherSectionGroup;
+
+            foreach (ConfigurationSection section in watcherSectionGroup.Sections)
+            {
+                if (section.GetType() == typeof(RuleSection))
+                {
+                    RuleSection ruleSection = (RuleSection)section;
+                    RuleElementCollection ruleElementCollection = ruleSection.Rules;
+                    foreach (RuleElement ruleElement in ruleElementCollection)
+                    {
+                        rules.Add(ruleElement);
+                    }
+                }
+
+                else if (section.GetType() == typeof(LocaleSection))
+                {
+                    LocaleSection localeSection = (LocaleSection)section;
+                    switch (localeSection.LocaleCode)
+                    {
+                        case "ru":
+                            currentCulture = new CultureInfo("ru-RU");
+                            break;
+                        default:
+                            currentCulture = new CultureInfo("en-US");
+                            break;
```

Comment:
For the sake of simplicity you could have the valid culture codes in configuration already. So that instead of this `switch` - `case` you would simply write:
```
currentCulture = new CultureInfo(localeSection.LocaleCode);
```

Reply:
I have applied this approach